components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.endpoint/src/gen/java/org/wso2/carbon/identity/configuration/mgt/endpoint/ResourceApi.java

```diff
@@ -0,0 +1,300 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.identity.configuration.mgt.endpoint;
+
+import io.swagger.annotations.ApiParam;
+import org.wso2.carbon.identity.configuration.mgt.endpoint.dto.AttributeDTO;
+import org.wso2.carbon.identity.configuration.mgt.endpoint.dto.ResourceAddDTO;
+import org.wso2.carbon.identity.configuration.mgt.endpoint.dto.ResourceDTO;
+import org.wso2.carbon.identity.configuration.mgt.endpoint.dto.ResourcesDTO;
+import org.wso2.carbon.identity.configuration.mgt.endpoint.factories.ResourceApiServiceFactory;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PATCH;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+@Path("/resource")
+@Consumes({"application/json"})
+@Produces({"application/json"})
+@io.swagger.annotations.Api(value = "/resource", description = "the resource API")
+public class ResourceApi {
+
+    private final ResourceApiService delegate = ResourceApiServiceFactory.getResourceApi();
+
+    @GET
+
+    @Consumes({"application/json"})
+    @Produces({"application/json"})
+    @io.swagger.annotations.ApiOperation(value = "Retrieve resources.\n", notes = "This API is used to retrieve resources.\n", response = ResourcesDTO.class)
+    @io.swagger.annotations.ApiResponses(value = {
+            @io.swagger.annotations.ApiResponse(code = 200, message = "Ok"),
+
+            @io.swagger.annotations.ApiResponse(code = 400, message = "Bad Request"),
+
+            @io.swagger.annotations.ApiResponse(code = 401, message = "Unauthorized"),
+
+            @io.swagger.annotations.ApiResponse(code = 500, message = "Server Error")})
+
+    public Response resourceGet() {
+
+        return delegate.resourceGet();
+    }
+
+    @PATCH
+    @Path("/{resource-type}")
+    @Consumes({"application/json"})
+    @Produces({"application/json"})
+    @io.swagger.annotations.ApiOperation(value = "Update existing resource\n", notes = "This API is used to update an existing resource given by the user.\n", response = ResourceDTO.class)
+    @io.swagger.annotations.ApiResponses(value = {
+            @io.swagger.annotations.ApiResponse(code = 200, message = "Ok"),
+
+            @io.swagger.annotations.ApiResponse(code = 400, message = "Bad Request"),
+
+            @io.swagger.annotations.ApiResponse(code = 401, message = "Unauthorized"),
+
+            @io.swagger.annotations.ApiResponse(code = 404, message = "Not Found"),
+
+            @io.swagger.annotations.ApiResponse(code = 500, message = "Server Error")})
+
+    public Response resourceResourceTypePatch(@ApiParam(value = "This represents the type of the resource to be added and can either be the name or id.", required = true) @PathParam("resource-type") String resourceType,
```

Comment:
exceeds 120 character length. please check other places also.

Reply:
Intentionally left due to a generated class.